src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -572,7 +564,7 @@ int FreezeBase::size_if_fast_freeze_available() {
   // so we subtract it only if we overlap with the caller, i.e. the current chunk isn't empty.
   // Consider leaving the chunk's argsize set when emptying it and removing the following branch,
   // although that would require changing stackChunkOopDesc::is_empty
-  if (chunk_sp < chunk->stack_size()) {
+  if (!chunk->is_empty()) {
```

Comment:
Seems you have implemented the suggestion in the comment so we can remove this branch and unconditionally decrement total_size_needed.

Reply:
I currently have an assert that checks that you shouldn't be asking for the argsize() if the chunk is empty, because it is so error prone. I think I'd like to keep the assert though - it was quite useful.

Reply:
We should be okay since _cont.argsize gets it from the ContinuationEntry. I tested it a bit and we would also need to update _fast_freeze_size to be cont_size() in the chunk empty case before calling freeze_fast_copy() otherwise we hit an assert there. But I can do this in another RFR if you want.

Reply:
That would be great, thank you!